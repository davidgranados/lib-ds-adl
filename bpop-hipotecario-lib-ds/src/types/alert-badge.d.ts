import { ReactNode } from "react";
import { Testable } from "./global";

export interface AlertBadgeProps extends Testable {
  children?: ReactNode;
  type?: "info" | "warning" | "success" | "danger";
}
