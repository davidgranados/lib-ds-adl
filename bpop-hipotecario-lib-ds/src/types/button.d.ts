import { HTMLPropsFor } from "./component";

export type ButtonVariant = "primary" | "secondary";
export type ButtonTextSize = "sm" | "md" | "lg";
export type ButtonSize = "xs" | ButtonTextSize;

export interface DefaultButtonProps extends HTMLPropsFor<"button"> {
  label: string;
  variant?: ButtonVariant;
  disabled?: boolean;
  onClick?: () => void;
}

export interface ButtonProps extends DefaultButtonProps {
  size?: ButtonSize;
}

export interface ButtonTextProps extends DefaultButtonProps {
  size?: ButtonTextSize;
}
