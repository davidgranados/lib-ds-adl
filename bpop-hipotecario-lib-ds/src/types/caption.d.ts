import { ReactNode } from "react";
import { Testable } from "./global";

export declare type InputFieldCaptionStatus = "default" | "error" | "warning" | "success";

export declare interface InputFieldCaptionProps extends Testable {
  children: ReactNode;
  status?: InputFieldCaptionStatus;
}
