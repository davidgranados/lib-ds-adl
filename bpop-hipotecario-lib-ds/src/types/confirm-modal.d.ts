import { Testable } from "./global";

export declare interface ConfirmModalProps extends Testable {
  title: string;
  description?: string;
  show: boolean;
  mainButtonText?: string;
  secondaryButtonText?: string;
  onMainButtonClick?: () => void;
  onSecondaryButtonClick?: () => void;
  onToggle?: (state: boolean) => void;
}
