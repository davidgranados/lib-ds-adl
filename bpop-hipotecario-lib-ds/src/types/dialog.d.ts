import { Testable } from "./global";

export declare interface DialogProps extends Testable {
  show: boolean;
  width?: number;
  onToggle?: (state: boolean) => void;
}
