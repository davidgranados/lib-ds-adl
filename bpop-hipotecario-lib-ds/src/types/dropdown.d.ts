import { Testable } from "./global";

export declare interface DropdownProps extends Testable {
  label: string;
  items: { label: string; value: string | number }[];
  disabled?: boolean;
  hasError?: boolean;
  caption?: string;
  placeholder?: string;
  value?: string | number;
  onChange: (v: string | number) => void;
}
