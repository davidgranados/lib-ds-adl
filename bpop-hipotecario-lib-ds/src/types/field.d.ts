import { ChangeEventHandler, FocusEventHandler, FormEventHandler } from "react";

import { InputBaseSize, InputBaseStatus } from "../../atoms/input-base";
import { TypographyStatus } from "../../atoms/typography";

export interface DefaultFieldProps {
  id: string;
  label: string;
  value?: string | number;
  status?: InputBaseStatus | TypographyStatus;
  caption?: string;
  placeholder?: string;
  size?: InputBaseSize;
  disabled?: boolean;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  onFocus?: FocusEventHandler<HTMLInputElement>;
  onBlur?: FocusEventHandler<HTMLInputElement>;
  onInput?: FormEventHandler<HTMLInputElement>;
}
