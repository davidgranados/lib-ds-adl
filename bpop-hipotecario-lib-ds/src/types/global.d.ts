import React from "react";

export type HTMLPropsFor<Tag extends keyof JSX.IntrinsicElements> = JSX.IntrinsicElements[Tag];
export type CombinedRef = Array<React.MutableRefObject<null> | React.ForwardedRef<HTMLInputElement>>;
