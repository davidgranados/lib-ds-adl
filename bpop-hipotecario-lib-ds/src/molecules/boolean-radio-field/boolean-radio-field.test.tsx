import React from "react";
import { mount, ReactWrapper } from "enzyme";

import { BooleanRadioFieldProps } from "../../types";

import BooleanRadioField from "./boolean-radio-field";

describe("<BooleanRadioField />", () => {
  let wrapper: ReactWrapper<typeof BooleanRadioField>;
  const props: BooleanRadioFieldProps = {
    id: "text-field-base-story",
    withTooltip: true,
    name: "international-payments",
    initialValue: false,
    labelText: "¿Realiza operaciones internacionales?",
    tooltipTextBody:
      "Son los movimientos financieros que realizas en otros países a través de cuentas en el exterior a tu nombre. No son compras por internet, suscripciones a servicios o remesas/giros.",
    onChange: (event) => console.log(event),
  };

  beforeEach(() => {
    jest.clearAllMocks();
    // given
    wrapper = mount<typeof BooleanRadioField>(<BooleanRadioField {...props} />);
  });
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
