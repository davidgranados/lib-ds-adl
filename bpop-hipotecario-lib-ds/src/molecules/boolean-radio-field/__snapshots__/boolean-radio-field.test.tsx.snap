// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BooleanRadioField /> should match snapshot 1`] = `
<div
  className="wrapper"
  data-testid="text-field-base-story-wrapper"
  id="text-field-base-story-wrapper"
>
  <div
    className="label"
  >
    ¿Realiza operaciones internacionales?
    Array [
      <span
        className="mobile-inline-only"
        data-testid="text-field-base-story-tooltip-icon-mobile-span-wrapper"
        id="text-field-base-story-tooltip-icon-mobile-span-wrapper"
      >
        <div
          className="wrapper"
          data-testid="text-field-base-story-tooltip-icon-mobile-wrapper"
          id="text-field-base-story-tooltip-icon-mobile-wrapper"
          onClick={[Function]}
        >
          <div
            className="content-wrapper"
            data-testid="text-field-base-story-tooltip-icon-mobile"
            id="text-field-base-story-tooltip-icon-mobile"
            style={
              Object {
                "width": "calc(calc(100vw - 0px) - 0px)",
              }
            }
          >
            <div
              className="content-body"
            >
              <span
                className="icon-wrapper"
              >
                <span
                  className="icon icon-close icon-close"
                  data-testid="text-field-base-story-tooltip-icon-mobile-icon-close"
                  id="text-field-base-story-tooltip-icon-mobile-icon-close"
                />
              </span>
              <div
                className="content-text"
              >
                <span>
                  Son los movimientos financieros que realizas en otros países a través de cuentas en el exterior a tu nombre. No son compras por internet, suscripciones a servicios o remesas/giros.
                </span>
              </div>
            </div>
          </div>
          <span
            className="icon icon-exclamation-open-circle trigger"
            data-testid="text-field-base-story-tooltip-icon-mobile-trigger"
            id="text-field-base-story-tooltip-icon-mobile-trigger"
          />
        </div>
      </span>,
      <span
        className="desktop-inline-only"
        data-testid="text-field-base-story-tooltip-icon-desktop-span-wrapper"
        id="text-field-base-story-tooltip-icon-desktop-span-wrapper"
      >
        <div
          className="wrapper"
          data-testid="text-field-base-story-tooltip-icon-desktop-wrapper"
          id="text-field-base-story-tooltip-icon-desktop-wrapper"
          onClick={[Function]}
        >
          <div
            className="content-wrapper content-wrapper--static-width"
            data-testid="text-field-base-story-tooltip-icon-desktop"
            id="text-field-base-story-tooltip-icon-desktop"
          >
            <div
              className="content-body"
            >
              <span
                className="icon-wrapper"
              >
                <span
                  className="icon icon-close icon-close"
                  data-testid="text-field-base-story-tooltip-icon-desktop-icon-close"
                  id="text-field-base-story-tooltip-icon-desktop-icon-close"
                />
              </span>
              <div
                className="content-text"
              >
                <span>
                  Son los movimientos financieros que realizas en otros países a través de cuentas en el exterior a tu nombre. No son compras por internet, suscripciones a servicios o remesas/giros.
                </span>
              </div>
            </div>
          </div>
          <span
            className="icon icon-exclamation-open-circle trigger"
            data-testid="text-field-base-story-tooltip-icon-desktop-trigger"
            id="text-field-base-story-tooltip-icon-desktop-trigger"
          />
        </div>
      </span>,
    ]
  </div>
  <div
    className="radio-buttons-wrapper"
  >
    <label
      className="wrapper wrapper--enabled"
      data-testid="text-field-base-story-false-radio-wrapper"
      id="text-field-base-story-false-radio-wrapper"
    >
      <span
        className="control-wrapper"
      >
        <input
          className="input"
          data-testid="text-field-base-story-false-radio"
          defaultChecked={true}
          id="text-field-base-story-false-radio"
          name="international-payments"
          onChange={[Function]}
          type="radio"
          value="false"
        />
        <span
          className="control"
        >
          <span
            className="control__circle"
          />
        </span>
      </span>
      <span
        className="label"
      >
        No
      </span>
    </label>
    <label
      className="wrapper wrapper--enabled"
      data-testid="text-field-base-story-true-radio-wrapper"
      id="text-field-base-story-true-radio-wrapper"
    >
      <span
        className="control-wrapper"
      >
        <input
          className="input"
          data-testid="text-field-base-story-true-radio"
          defaultChecked={false}
          id="text-field-base-story-true-radio"
          name="international-payments"
          onChange={[Function]}
          type="radio"
          value="true"
        />
        <span
          className="control"
        >
          <span
            className="control__circle"
          />
        </span>
      </span>
      <span
        className="label"
      >
        Si
      </span>
    </label>
  </div>
</div>
`;
